Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -25,7 +25,7 @@
 namespace DiscIO
 {
 static u32 ComputeNameSize(const File::FSTEntry& parent_entry);
-static std::string ASCIIToLowercase(std::string str);
+static std::string ASCIIToUppercase(std::string str);
```

Comment:
At the beginning I was thinking of a program comparing the disc content with the SD card but you're right I think it's doable and easier on computer than implementing a proper hardware test.

As you pointed out earlier, that's not the standard way of doing a "case-insensitive" comparisons according to the strcasecmp manual. The issue ds84182 has is related to underscore, that's why I think the solution might be simpler and translating it to uppercase might introduce other issues. Typically in that situation, if I had to throw a random guess, I would just say that issue can probably be solved with Windows _stricmp function (which isn't available on Linux, IIRC).

> The characters from 91 to 96 in the ASCII table ('[', '\', ']', '^', '_', and '`') evaluate as less than any alphabetic character. This ordering is identical to that of stricmp.